context.go

```diff
@@ -0,0 +1,162 @@
+	Page      PageInfo               `json:"page,omitempty"`
+	Referrer  ReferrerInfo           `json:"referrer,omitempty"`
+	Screen    ScreenInfo             `json:"screen,omitempty"`
+	IP        net.IP                 `json:"ip,omitempty"`
+	Locale    string                 `json:"locale,omitempty"`
+	Timezone  string                 `json:"timezone,omitempty"`
+	UserAgent string                 `json:"userAgent,omitempty"`
+	Traits    map[string]interface{} `json:"traits,omitempty"`
+
+	// This map is used to allow extensions to the context specifications that
+	// may not be documented or could be introduced in the future.
+	// The fields of this map are inlined in the serialized context object,
+	// there is no actual "extra" field in the JSON representation.
+	Extra map[string]interface{} `json:"-"`
+}
+
+// This type provides the representation of the `context.app` object as defined
+// in https://segment.com/docs/spec/common/#context
+type AppInfo struct {
+	Name    string `json:"name,omitempty"`
+	Version string `json:"version,omitempty"`
+	Build   string `json:"build,omitempty"`
+}
+
+// This type provides the representation of the `context.campaign` object as
+// defined in https://segment.com/docs/spec/common/#context
+type CampaignInfo struct {
+	Name    string `json:"name,omitempty"`
+	Source  string `json:"source,omitempty"`
+	Medium  string `json:"medium,omitempty"`
+	Term    string `json:"term,omitempty"`
+	Content string `json:"content,omitempty"`
+}
+
+// This type provides the representation of the `context.device` object as
+// defined in https://segment.com/docs/spec/common/#context
+type DeviceInfo struct {
+	Id           string `json:"id,omitempty"`
+	Manufacturer string `json:"manufacturer,omitempty"`
+	Model        string `json:"model,omitempty"`
+	Name         string `json:"name,omitempty"`
+	Type         string `json:"type,omitempty"`
+	Version      string `json:"version,omitempty"`
+}
+
+// This type provides the representation of the `context.library` object as
+// defined in https://segment.com/docs/spec/common/#context
+type LibraryInfo struct {
+	Name    string `json:"name,omitempty"`
+	Version string `json:"version,omitempty"`
+}
+
+// This type provides the representation of the `context.location` object as
+// defined in https://segment.com/docs/spec/common/#context
+type LocationInfo struct {
+	City      string  `json:"city,omitempty"`
+	Country   string  `json:"country,omitempty"`
+	Region    string  `json:"region,omitempty"`
+	Latitude  float64 `json:"latitude,omitempty"`
+	Longitude float64 `json:"longitude,omitempty"`
+	Speed     float64 `json:"speed,omitempty"`
+}
+
+// This type provides the representation of the `context.network` object as
+// defined in https://segment.com/docs/spec/common/#context
+type NetworkInfo struct {
+	Bluetooth bool   `json:"bluetooth,omitempty"`
+	Cellular  bool   `json:"cellular,omitempty"`
+	WIFI      bool   `json:"wifi,omitempty"`
```

Comment:
Here I'm not sure about the data type, my understanding is that it should be booleans but the documentation isn't very clear about it: https://segment.com/docs/spec/common/#context